dropgz/pkg/embed/payload.go

```diff
@@ -86,6 +87,17 @@ func deploy(src, dest string) error {
 		return err
 	}
 	defer rc.Close()
+	if _, err := os.Stat(dest); err == nil || !os.IsNotExist(err) {
+		old_file_dest := dest + OLD_FILE_SUFFIX
+		if err = os.RemoveAll(old_file_dest); err != nil {
+			return errors.Wrapf(err, "not able to remove the %s", &old_file_dest)
+		}
+		if err = os.Rename(dest, old_file_dest); err != nil {
+			return errors.Wrapf(err, "not able to rename the %s to %s", dest, old_file_dest)
+		}
```

Comment:
I think the `os.RemoveAll` is unnecessary - [`os.Rename`](https://pkg.go.dev/os#Rename) will remove the file at dest, if it exists.

Reply:
For the first the file won't be present, so `os.Remove` will error out.

Reply:
I didn't say anything about `os.Remove`